src/jltypes.c

```diff
@@ -3760,8 +3760,13 @@ void jl_init_types(void) JL_GC_DISABLED
                                         "_state",
                                         "sticky",
                                         "_isexception",
-                                        "priority"),
-                        jl_svec(16,
+                                        "priority",
+                                        "is_timing_enabled",
+                                        "first_enqueued_at",
+                                        "last_dequeued_at",
+                                        "cpu_time_ns",
+                                        "wall_time_ns"),
```

Comment:
Should these go in a separate struct called `metrics` or something @vtjnash? We won't be able to initialize that struct type before this, so it would have to be `jl_any_type`, but that would mean we'd have to alloc this in `jl_new_task`. I don't know how/if we can do an embedded struct here?

Reply:
It could be a separate struct inlined here, but I don't see much value to that. More valuable would probably be marking each of these as atomic though (in jltypes.c) so that observing them isn't UB. Most reads can probably be monotonic, but some writes are probably release and some reads might be acquires

Reply:
Okay on the separation concern. And yeah, I guess it is necessary for these to be atomic. Will discuss with @nickrobinson251.

Reply:
Changed to atomics. But i wasn't sure from the docs how to select the correct memory ordering (so you put `:monotonic` everywhere for now) -- would love advice

Reply:
@vtjnash would you be able to take a look and advise on which orderings to use for the various atomic reads and writes in the PR?